tests/sentry/nodestore/bigtable/backend/tests.py

```diff
@@ -103,28 +103,6 @@ def ns(request):
         yield MockedBigtableNodeStorage(project="test")
 
 
-@pytest.mark.parametrize(
-    "compression,expected_prefix",
-    [(True, (b"\x78\x01", b"\x78\x9c", b"\x78\xda")), (False, b"{"), ("zstd", b"\x28\xb5\x2f\xfd")],
-    ids=["zlib", "ident", "zstd"],
-)
-def test_get(ns, compression, expected_prefix):
-    ns.store.compression = compression
-    node_id = "node_id"
-    data = {"foo": "bar"}
-    ns.set(node_id, data)
-
-    # Make sure this value does not get used during read. We may have various
-    # forms of compression in bigtable.
-    ns.store.compression = lambda: 1 / 0
-    # Do not use cache as that entirely bypasses what we want to test here.
-    ns.cache = None
-    assert ns.get(node_id) == data
-
-    raw_data = ns.store._get_table().read_row("node_id").cells["x"][b"0"][0].value
-    assert raw_data.startswith(expected_prefix)
```

Comment:
This got moved to test `BigtableKVStorage` specifically and split into two tests:

1. One test that maintains the existing behavior of this test to ensure the value in the database is actually passed through the compression function (and also tests to make sure the flag is set correctly)
2. Another test that ensures that the storage instance can read values written with other compression strategies than the one it uses to write data (the behavior that was suggested here by setting `store.compression` to an invalid value but not actually tested.)